apps/artemis-web/HOC/withAuth.tsx

```diff
@@ -0,0 +1,74 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import { extractUser } from '../lib/helpers';
+import passport from '../lib/passport';
+import auth from '../middleware/auth';
+import { initializeApollo } from '../utils/graphql';
+import nc from 'next-connect';
+import React from 'react';
+import { useCookie } from 'next-cookie';
+import jwt from 'jsonwebtoken';
+
+const withAuth = (WrappedComponent, ACL = ['admin', 'user']) => {
+  class App extends React.PureComponent {
+    render() {
+      return <WrappedComponent {...this.props} />;
+    }
+
+    static async getInitialProps({ req, res }) {
+      if (req) {
+        const props = await getProps({ req });
+        if (props.user) {
```

Comment:
we don't need this.. if a page is not marked for authorization then we just simply don't redirect.